PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -0,0 +1,214 @@
+import com.microsoft.intellij.rxjava.IdeaSchedulers;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.sparkserverless.SparkServerlessADLAccountNode;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+
+public class SparkServerlessProvisionDialog extends DialogWrapper
+        implements SettableControl<SparkServerlessClusterProvisionSettingsModel> {
+    @NotNull
+    private SparkServerlessClusterProvisionCtrlProvider ctrlProvider;
+
+    @NotNull
+    private SparkServerlessADLAccountNode adlAccountNode;
+
+    private JTextField clusterNameField;
+    private JTextField adlAccountField;
+    private JTextField previousSparkEventsField;
+    private JTextField masterCoresField;
+    private JTextField masterMemoryField;
+    private JTextField workerCoresField;
+    private JTextField workerMemoryField;
+    private JTextField workerNumberOfContainersField;
+    private JTextField availableAUField;
+    private JTextField totalAUField;
+    private JTextField calculatedAUField;
+    private JLabel masterMemoryLabel;
+    private JLabel masterCoresLabel;
+    private JLabel clusterNameLabel;
+    private JLabel adlAccountLabel;
+    private JLabel previousSparkEventsLabel;
+    private JLabel availableAULabel;
+    private JLabel calculatedAULabel;
+    private JLabel workerCoresLabel;
+    private JLabel workerMemoryLabel;
+    private JLabel workerNumberOfContainersLabel;
+    private JTextField errorMessageField;
+    private JPanel provisionDialogPanel;
+    private JButton refreshButton;
+
+    public SparkServerlessProvisionDialog(@NotNull SparkServerlessADLAccountNode adlAccountNode) {
+        // TODO: canBeParent
+        super((Project)adlAccountNode.getProject(), true);
+
+        this.ctrlProvider = new SparkServerlessClusterProvisionCtrlProvider(
+                this, new IdeaSchedulers((Project)adlAccountNode.getProject()));
+        this.adlAccountNode = adlAccountNode;
+
+        init();
+        this.setTitle("Spark Serverless Cluster");
+        errorMessageField.setBackground(this.provisionDialogPanel.getBackground());
+        errorMessageField.setBorder(BorderFactory.createEmptyBorder());
+        this.setModal(true);
+
+        adlAccountField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                availableAUField.setText(String.valueOf(ctrlProvider.getAvailableAU(adlAccountField.getText())));
+                totalAUField.setText(String.valueOf(ctrlProvider.getTotalAU(adlAccountField.getText())));
+            }
+        });
+        refreshButton.addActionListener(e -> {
+                availableAUField.setText(String.valueOf(ctrlProvider.getAvailableAU(adlAccountField.getText())));
+                totalAUField.setText(String.valueOf(ctrlProvider.getTotalAU(adlAccountField.getText())));
```

Comment:
>totalAUField.setText(String.valueOf(ctrlProvider.getTotalAU(adlAccountField.getText()))); [](start = 16, length = 89)

This will not change, no need to refresh